### PR TITLE
Factor mount utility code out gce_pd volume plugin

### PR DESCRIPTION
**Pull request by pmorie**
Currently the gce_pd volume plugin contains a significant amount of utility code that should be consolidated under `pkg/util/mount`.

@thockin @markturansky

### CODE REVIEW

**Comment by pmorie** [comment]
@calfonso @rootfs I believe this is relevant to your interests

**Comment by pmorie** [comment]
#4612 #4601

File: test/e2e/pd.go [hunk 1]
```diff
@@ -47,7 +47,8 @@ var _ = Describe("PD", func() {
 
 		nodes, err := c.Nodes().List()
 		expectNoError(err, "Failed to list nodes for e2e cluster.")
-		Expect(len(nodes.Items) >= 2)
+		fmt.Printf("Nodes: %v", len(nodes.Items))
```
Comment (markturansky) [inline test/e2e/pd.go]: Debug statement?

**Comment by pmorie** [comment]
@thockin I'm ready for review on this once you have some bandwidth.

File: pkg/util/mount/mount.go [hunk 2]
```diff
@@ -49,3 +49,30 @@ type MountPoint struct {
 	Freq   int
 	Pass   int
 }
+
+// Examines /proc/mounts to find all other references to the device referenced
```
Comment (rootfs) [inline pkg/util/mount/mount.go]: why not just create a mount_util.go?

Comment (pmorie) [inline pkg/util/mount/mount.go]: @rootfs things are in the files they are to ensure that the build flags for
each function / type didn't change across this commit.

File: pkg/util/mount/fake.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// +build linux
```
Comment (smarterclayton) [inline pkg/util/mount/fake.go]: Why is this +build linux?

**Comment by smarterclayton** [comment]
One comment, otherwise LGTM

**Comment by pmorie** [comment]
@smarterclayton nit fixed, I had missed that when I copied the header in.

File: pkg/util/mount/fake.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+// FakeMounter implements mount.Interface.
+type FakeMounter struct {
+	mountPoints []MountPoint
```
Comment (thockin) [inline pkg/util/mount/fake.go]: How is one supposed to set the fake mount points if it is a private field?

File: pkg/util/mount/mount.go [hunk 5]
```diff
@@ -49,3 +49,30 @@ type MountPoint struct {
...
+// by mountPath.
```
Comment (thockin) [inline pkg/util/mount/mount.go]: Comment should detail that it returns a list of paths.

**Comment by thockin** [comment]
just a couple nits - sorry for late review, send me a followup PR and I'll fast-track it.

**Pull request merged**